control-plane-operator/controllers/hostedcontrolplane/ignitionserver/ignitionserver.go

```diff
@@ -188,6 +190,48 @@ func ReconcileIgnitionServer(ctx context.Context,
 	if hcp.Spec.Platform.Type != hyperv1.IBMCloudPlatform {
 		servingCertSecretName = manifests.IgnitionServerCertSecret("").Name
 	}
+
+	if len(openShiftRegistryOverrides) > 0 {
+		// Ignition server cannot handle ImageContentSourcePolicies or ImageDigestMachineSet, so we need to
+		// fill the registryOverrides in the case of disconnected environments
+		mirrorImage := lookupDisconnectedRegistry(ctx, openShiftRegistryOverrides)
+		if len(mirrorImage) < 1 {
+			// In a weird case where the entry is not in the openShiftRegistryOverrides, we need to figure it out
+			log.Info("failed to find the release entry in the openShiftRegistryOverrides, figuring out the disconnected registry")
+			sourceRef, err := reference.Parse(hcp.Spec.ReleaseImage)
+			if err != nil {
+				return fmt.Errorf("failed to parse hosted control plane image reference %q: %w", hcp.Spec.ReleaseImage, err)
+			}
+			privateRegistry := sourceRef.Registry
+			if strings.Contains(privateRegistry, "quay.io") || strings.Contains(privateRegistry, "registry.redhat.io") {
+				return fmt.Errorf("failed to reconcile ignition server, the ReleaseImage set should point to a disconnected registry, now is pointing to: %s", privateRegistry)
+			}
+			mirrorImage = fmt.Sprintf("%s/openshift/release", privateRegistry)
+		}
+
+		privateRegistry := strings.Split(mirrorImage, "/")[0]
+
+		if !strings.HasPrefix(componentImages["machine-config-operator"], privateRegistry) {
+			ocpReleaseMcoImage := componentImages["machine-config-operator"]
+			mcoSha, err := splitSha(ocpReleaseMcoImage)
+			if err != nil {
+				return fmt.Errorf("failed extracting image sha %s into ignition server reconciliation: %w", ocpReleaseMcoImage, err)
+			}
+			registryOverrides[ocpReleaseMcoImage] = fmt.Sprintf("%s@%s", mirrorImage, mcoSha)
+		}
+
+		if !strings.HasPrefix(componentImages["cluster-config-operator"], privateRegistry) {
+			ocpReleaseCCOImage := componentImages["cluster-config-operator"]
+			ccoSha, err := splitSha(ocpReleaseCCOImage)
```

Comment:
You can also do this with reference.Parse() ... the ref.ID is the sha

Reply:
I've change all the references to SplitSha function (tests included) ty for the suggestion